architecture-building-systems/CityEnergyAnalyst
# Adding concept algorithms, building models developed by TUM into CEA

Algorithms developed as part of the CONCEPT project are integrated into CEA in this PR.

Along with this the building model developed by TUMCREATE is also integrated into CEA in this PR. Currently, TUMCREATE are in the process of changing their code into MIT license (almost done), if that finishes, there will be minor changes in this part.

In `cea\concept_project`, there are 4 separate folders namely: `algorithm_planning`, `algorithm_planning_and_operation`, `model_building`, `data`. There is also a `README.md` file which describes the prerequisites for running the algorithms.

The `cea\concept_project\data` folder has the examples to be tested (in the folder `IBPSA`). The results generated will be saved in the folder `cea\concept_project\data\output_operation` (after running `cea\concept_project\algorithm_operation\operation_main.py`). Similary in the folder `cea\concept_project\data\output_planning_and_operation` (after running `cea\concept_project\algorithm_planning_and_operation`).

The way to test the PR:
1. Install the prerequisites and dependencies
2. Go to `cea\concept_project\config.py` and set the `time_start` and `time_end` (test with time_start = 2005-01-01 00:00:00 and time_end = 2005-01-01 23:30:00, this runs MPC model for one day and should be very fast). After running for a day, you can increase the time frame to an entire year.
3. Run the `cea\concept_project\algorithm_operation\operation_main.py`
4. Run the `cea\concept_project\algorithm_planning_and_operation\planning_and_operation_main.py`
5. Check the results in the corresponding folders

Future Refactoring Areas:

1. De link the code to access the examples outside of the GitHub and save the results on the computer
2. Move the concept_project.config parameters to cea.config
3. Add in more documentation
